### ISSUE TITLE
Add new dispatchables to PalletXcm to allow remote locking 

### ISSUE THREAD

**Issue by IkerAlus** [paritytech/polkadot-sdk#3546]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

At the moment, chains implementing the PalletXcm do not allow their users to utilise the remote locking functionalities introduced in XCM v3.  There are no interfaces exposed directly to users to create XCM messages with the instructions `LockAsset`, `UnlockAsset`, etc.

### Request

Implement several dispatchables in the PalletXcm to allow the user journey for remote locking as explained in the examples of the XCM docs: https://paritytech.github.io/xcm-docs/journey/locks/locks.html

### Solution

 Following the original vision of[ XCM v3: Remote locking](https://github.com/paritytech/polkadot/issues/4808) the idea would be to implement two new dispatchables as:
 - `lock_asset(origin, asset: MultiAsset, unlocker: MultiLocation)`: Locks asset owned by origin with ID `(b"xcm_lock", unlocker).encode()`. Sends a `NoteAssetLocked` message to `unlocker` with `asset` and `origin` (converted to a `MultiLocation`).
- `unlock_asset(origin, asset: MultiAsset, locker: MultiLocation)`: Sends an `UnlockAsset` instruction to `locker` for `asset` IFF there is a registered remote lock for `asset` and it is currently unused.

### Are you willing to help with this request?

No.

**Comment by bkchr** [comment]
With allowing `execute_xcm` in `pallet-xcm` this should be solved? Aka you can just write a XCM message that does what you want to do?

**Comment by acatangiu** [comment]
> With allowing `execute_xcm` in `pallet-xcm` this should be solved? Aka you can just write a XCM message that does what you want to do?

Yes, exactly! We should focus on building (off-chain) tools to create these specialized XCMs that can then be dispatched on any XCM-enabled chain.

(I just opened an RFC describing the same idea for asset transfers https://github.com/paritytech/polkadot-sdk/issues/4736)
